app.py

```diff
@@ -33,6 +36,64 @@
 # Set the secret key to enable access to session data.
 app.secret_key = os.urandom(24)
 
+client = datastore.Client()
+
+GENIUS_ACCESS_TOKEN = os.getenv('GENIUS_ACCESS_TOKEN')
+
+@app.route('/home')
+def recommend():
+    """Recommend a song"""
+    headers = {
+        'Authorization': 'Bearer {token}'.format(token=GENIUS_ACCESS_TOKEN)}
+
+    # Hard-coding a song recommendation
+    url = 'https://api.genius.com/'
+    endpoint = 'songs/'
+
+    # Song ID for 'Space Song' by Beach House
+    song_id = '1929412'
+
+    response = requests.get(
+        url + endpoint + song_id,
+        headers=headers)
+
+    song = response.json()['response']['song']
+
+    return flask.render_template('index.html', song=song)
+
+@app.route('/likes')
+def get_likes():
+    """Get liked songs"""
+    kind = 'LikedSong'
+
+    query = client.query(kind=kind)
+    songs = list(query.fetch())
+
+    return flask.render_template('likes.html', songs=songs)
+
+@app.route('/like', methods=['POST'])
+def add_like():
+    """Like a song"""
+    if flask.request.method == 'POST':
+        song = ast.literal_eval(flask.request.form['song'])
+
+        with client.transaction():
+            kind = 'LikedSong'
+
+            likedsong = datastore.Entity(client.key(kind))
+            likedsong.update({
+                'album': song['album']['name'],
+                'apple_music_player_url': song['apple_music_player_url'],
+                'artist': song['primary_artist']['name'],
+                'embed_content': song['embed_content'],
+                'song_art_image_thumbnail_url': song['song_art_image_thumbnail_url'],
+                'title': song['title']})
+
+            client.put(likedsong)
+
+    return flask.redirect(flask.url_for('get_likes'))
```

Comment:
this should be returning JSON, and using angular routing to handle the redirects